### DISCUSSION TITLE
Is it possible to provide syntactic sugar for the for loop, similar to the for loop in GDScript in Godot?

### DISCUSSION THREAD

**Discussion by paxhujing** [dotnet/runtime#109668] [General]
[https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_advanced.html#doc-gdscript-more-efficiently](url)

`for s in strings:`
`  print(s)`

`for key in dict:`
`  print(key, " -> ", dict[key)`

`for i in range(strings.size()):`
`  print(strings[i])`

**Comment by huoyaoyuan** [comment]
Syntax discussion goes to https://github.com/dotnet/csharplang/discussions . There is `foreach` loop since C# 1.

**Reply by paxhujing** [reply]
Use
`for var i in array.Length`
'  print(array[i])' 
instead of
`for(var i=0;i<array.Length;i++)`
'  print(array[i])' 

or

`for var i in (1, 2, array.Length)`
'  print(array[i])' 
instead of
`for(var i=1;i<array.Length;i+=2)`
'  print(array[i])' 

not foreach iterator

**Reply by huoyaoyuan** [reply]
See https://github.com/dotnet/csharplang/discussions/8318 , https://github.com/dotnet/csharplang/discussions/1584 and https://github.com/dotnet/csharplang/discussions/3939 .
Enumerator is well customizable for such usage. There is no interest for new syntax. You can `global using` or `using static` your own ones for shortcuts.
